backup/restore phases for async plugin operations

sseago opened the pull request:
Thank you for contributing to Velero!

# Please add a summary of your change
Adds new backup and restore phases to support async plugin operations:
- WaitingForPluginOperations
- WaitingForPluginOperationsPartiallyFailed

# Does your change fix a particular issue?

Fixes #5491 #5492

# Please indicate you've done the following:

- [x ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [x ] Updated the corresponding documentation in `site/content/docs/main`.

anshulahuja98 commented:
Hi @sseago, are there plans to also expose the list/names of plugins on which it is waiting?
Otherwise from the status it is difficult to debug which plugin is taking time or could potentially be down.
Is that part of intended design - if not does it make sense to add names?

sseago commented:
@anshulahuja98 Yes -- this is probably something we'll end up putting in the `velero describe` output for backup/restore. `describe` already pulls from object storage to get warning/error, so this would be one more thing to pull in. That way `velero describe backup --details` would list the operations and their current status/progress.

sseago commented:
@anshulahuja98 See https://github.com/vmware-tanzu/velero/pull/5700 -- the information available for each operation will include the following:
plugin, item ID, operation ID, and progress information for that operation.

shubham-pampattiwar commented on site/content/docs/main/api-types/schedule.md:
```diff
@@ -143,7 +143,9 @@ spec:
+  # The current phase of the latest scheduled backup.
+  # Valid values are New, FailedValidation, InProgress, WaitingForPluginOperations,
+  # WaitingForPluginOperationsPartiallyFailed, Completed, PartiallyFailed, Failed.
   phase: ""
```
@sseago I believe this phase is the current phase of the schedule and not backup. Schedule phase is different from backup phase https://github.com/vmware-tanzu/velero/blob/main/pkg/apis/velero/v1/schedule_types.go#L67

sseago replied:
@shubham-pampattiwar Oh, you're right The doc is wrong here -- it says that phase is the phase of the backup, but it's not, it's hte schedule phase. I'll update the comments to reflect this.

sseago commented:
@blackpiglet @shubham-pampattiwar I fixed a problem with the schedule docs. The prior content was incorrect, listing Backup phases (which I modified in the first iteration of this PR), but schedules have their own independent list of phases, so I've updated the schedule doc to reflect that. If you could review again with the changes, that would be great.

shubham-pampattiwar reviewed (approved):
Thanks @sseago LGTM!
